lldb/source/Plugins/Language/Swift/SwiftLanguage.cpp

```diff
@@ -1074,9 +1082,25 @@ SwiftLanguage::GetHardcodedSynthetics() {
 
       casted_to_swift->SetName(ConstString("Swift_Type"));
 
-      SyntheticChildrenSP synth_sp =
-          SyntheticChildrenSP(new ValueObjectWrapperSyntheticChildren(
-              casted_to_swift, SyntheticChildren::Flags()));
+      SyntheticChildrenSP synth_sp;
+      if (should_wrap_in_ptr) {
+        // If we have a pointer to a Swift value type, dereference the pointer
+        // and present those as the contents instead.
+        auto children = lldb_private::formatters::swift::
+            ExtractChildrenFromSwiftPointerValueObject(casted_to_swift);
+
+        if (children.empty())
+          return nullptr;
+        // The pointer should only have one child: the pointee.
+        assert(children.size() == 1 &&
+               "Unexpected size for pointer's children!");
+
+        synth_sp = SyntheticChildrenSP(new ValueObjectWrapperSyntheticChildren(
+            children[0], SyntheticChildren::Flags()));
+      } else {
+        synth_sp = SyntheticChildrenSP(new ValueObjectWrapperSyntheticChildren(
```

Comment:
can you sink this expression down and make it take an _sp variable that's set in the if-then-else?